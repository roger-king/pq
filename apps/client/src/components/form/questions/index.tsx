export * from './option.form';
export * from './question.form';
