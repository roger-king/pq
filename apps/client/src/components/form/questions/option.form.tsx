import { Box, Button, CheckBox, TextInput } from 'grommet';
import { SubtractCircle } from 'grommet-icons';
import React from 'react';

export interface OptionFormProps {
  showRemove: boolean;
  currentIndex: number;
  isAnswer: boolean;
  answerOnChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  title: string;
  titleOnChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  remove: () => void;
}

export const OptionForm: React.FC<OptionFormProps> = ({
  isAnswer,
  title,
  answerOnChange,
  titleOnChange,
  currentIndex,
  showRemove,
  remove,
}: OptionFormProps) => {
  return (
    <Box direction="row" gap="small" align="center">
      <CheckBox value={currentIndex} label="answer" reverse checked={isAnswer} onChange={answerOnChange} />
      <TextInput placeholder={`Option ${currentIndex + 1}`} value={title} onChange={titleOnChange} />
      {showRemove && <Button icon={<SubtractCircle />} onClick={remove} />}
    </Box>
  );
};
