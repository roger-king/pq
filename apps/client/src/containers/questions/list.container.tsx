import React from 'react';
import { Box, Heading, Text } from 'grommet';
import { useQuery } from 'react-query';
import Axios from 'axios';
import { Question } from '../../@types';
import { API_URL } from '../../constants';

export interface QuestionListProps {
  gameId: number;
}

export const QuestionList: React.FC<QuestionListProps> = ({ gameId }) => {
  const { data: questions, isLoading } = useQuery<{ data: Question[] }>(`game_${gameId}_questions`, async () => {
    const d = Axios.get(`${API_URL}/games/${gameId}/questions`);
    return d;
  });

  if (isLoading && !questions) {
    return <Box>loading...</Box>;
  }

  if (questions) {
    const { data } = questions;

    return (
      <Box margin="small" width="60%" height="400px">
        {data.map((q, i) => (
          <Box key={`${q.q}`} elevation="small" pad="medium">
            <Heading level="3">
              {i + 1}. {q.q}
            </Heading>
            {q.options.map((o) => (
              <Box key={o.key} gap="small" direction="row">
                <Text size="1.5em">
                  <b>{o.key}:</b>
                </Text>
                <Text size="1.5em">{o.title}</Text>
              </Box>
            ))}
          </Box>
        ))}
      </Box>
    );
  }

  return <Box>Something went wrong getting questions. We are looking into it.</Box>;
};
