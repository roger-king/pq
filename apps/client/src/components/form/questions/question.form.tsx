import React from 'react';
import { Box, Button, TextInput } from 'grommet';
import { AddCircle } from 'grommet-icons';

import { QuestionCard } from './../../card';
import { OptionFormProps, OptionForm } from './option.form';

export interface QuestionFormProps {
  qIndex: number;
  q: string;
  qOnChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  options: OptionFormProps[];
  addOption: () => void;
  maxNumOptions: number;
}

export const QuestionForm: React.FC<QuestionFormProps> = ({
  q,
  qOnChange,
  qIndex,
  options,
  addOption,
  maxNumOptions,
}: QuestionFormProps) => {
  return (
    <QuestionCard elevation={true}>
      <TextInput placeholder={`Question ${qIndex + 1}`} value={q} onChange={qOnChange} />
      <Box gap="small" margin="small">
        {options.map((o) => (
          <OptionForm key={`question-option-form-${o.currentIndex}`} {...o} />
        ))}
        {options.length < maxNumOptions && <Button icon={<AddCircle />} onClick={addOption} />}
      </Box>
    </QuestionCard>
  );
};
