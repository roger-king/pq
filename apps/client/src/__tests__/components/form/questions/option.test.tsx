import React from 'react';
import renderer from 'react-test-renderer';

import { OptionForm } from '../../../../components/form/questions';

describe('<OptionForm />', () => {
  test('should display basic Options Form: not answer or remove not shown', async () => {
    const tree = renderer
      .create(
        <OptionForm
          showRemove={false}
          currentIndex={1}
          isAnswer={false}
          answerOnChange={(e) => console.log('testing', e)}
          title="test"
          titleOnChange={(e) => console.log('test', e)}
          remove={() => console.log('remove')}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should display basic Options Form: not answer or remove shown', async () => {
    const tree = renderer
      .create(
        <OptionForm
          showRemove
          currentIndex={1}
          isAnswer={false}
          answerOnChange={(e) => console.log('testing', e)}
          title="test"
          titleOnChange={(e) => console.log('test', e)}
          remove={() => console.log('remove')}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should display basic Options Form is answer or remove shown', async () => {
    const tree = renderer
      .create(
        <OptionForm
          showRemove
          currentIndex={1}
          isAnswer
          answerOnChange={(e) => console.log('testing', e)}
          title="test"
          titleOnChange={(e) => console.log('test', e)}
          remove={() => console.log('remove')}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
