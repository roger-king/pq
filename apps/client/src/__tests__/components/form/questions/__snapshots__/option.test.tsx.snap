// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptionForm /> should display basic Options Form is answer or remove shown 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 dfYOnT"
>
  <label
    checked={true}
    className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 htOOow"
    onClick={[Function]}
  >
    <span>
      answer
    </span>
    <div
      checked={true}
      className="StyledBox-sc-13pk1d4-0 jUHgwK StyledCheckBox-sc-1dbk5ju-6 hOFfoi"
    >
      <input
        checked={true}
        className="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 fYuA-Ds"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
        value={1}
      />
      <div
        checked={true}
        className="StyledBox-sc-13pk1d4-0 icgjqF StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 kRWCTA"
      >
        <svg
          checked={true}
          className="StyledCheckBox__StyledCheckBoxIcon-sc-1dbk5ju-0 uDaMI"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <path
            d="M6,11.3 L10.3,16 L18,6.2"
            fill="none"
          />
        </svg>
      </div>
    </div>
  </label>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 eBVJAG"
  />
  <div
    className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
  >
    <input
      autoComplete="off"
      className="StyledTextInput-sc-1x30a0s-0 fwbQtl"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="Option 2"
      value="test"
    />
  </div>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 eBVJAG"
  />
  <button
    className="StyledButton-sc-323bzc-0 hiyJPE"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    type="button"
  >
    <svg
      aria-label="SubtractCircle"
      className="StyledIcon-ofa7kd-0 dEeaTZ"
      viewBox="0 0 24 24"
    >
      <path
        d="M12,22 C17.5228475,22 22,17.5228475 22,12 C22,6.4771525 17.5228475,2 12,2 C6.4771525,2 2,6.4771525 2,12 C2,17.5228475 6.4771525,22 12,22 Z M6,12 L18,12"
        fill="none"
        stroke="#000"
        strokeWidth="2"
      />
    </svg>
  </button>
</div>
`;

exports[`<OptionForm /> should display basic Options Form: not answer or remove not shown 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 dfYOnT"
>
  <label
    checked={false}
    className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 htOOow"
    onClick={[Function]}
  >
    <span>
      answer
    </span>
    <div
      checked={false}
      className="StyledBox-sc-13pk1d4-0 jUHgwK StyledCheckBox-sc-1dbk5ju-6 hOFfoi"
    >
      <input
        checked={false}
        className="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 fYuA-Ds"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
        value={1}
      />
      <div
        checked={false}
        className="StyledBox-sc-13pk1d4-0 jYPXhZ StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 kRWCTA"
      />
    </div>
  </label>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 eBVJAG"
  />
  <div
    className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
  >
    <input
      autoComplete="off"
      className="StyledTextInput-sc-1x30a0s-0 fwbQtl"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="Option 2"
      value="test"
    />
  </div>
</div>
`;

exports[`<OptionForm /> should display basic Options Form: not answer or remove shown 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 dfYOnT"
>
  <label
    checked={false}
    className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 htOOow"
    onClick={[Function]}
  >
    <span>
      answer
    </span>
    <div
      checked={false}
      className="StyledBox-sc-13pk1d4-0 jUHgwK StyledCheckBox-sc-1dbk5ju-6 hOFfoi"
    >
      <input
        checked={false}
        className="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 fYuA-Ds"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
        value={1}
      />
      <div
        checked={false}
        className="StyledBox-sc-13pk1d4-0 jYPXhZ StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 kRWCTA"
      />
    </div>
  </label>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 eBVJAG"
  />
  <div
    className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
  >
    <input
      autoComplete="off"
      className="StyledTextInput-sc-1x30a0s-0 fwbQtl"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="Option 2"
      value="test"
    />
  </div>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 eBVJAG"
  />
  <button
    className="StyledButton-sc-323bzc-0 hiyJPE"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    type="button"
  >
    <svg
      aria-label="SubtractCircle"
      className="StyledIcon-ofa7kd-0 dEeaTZ"
      viewBox="0 0 24 24"
    >
      <path
        d="M12,22 C17.5228475,22 22,17.5228475 22,12 C22,6.4771525 17.5228475,2 12,2 C6.4771525,2 2,6.4771525 2,12 C2,17.5228475 6.4771525,22 12,22 Z M6,12 L18,12"
        fill="none"
        stroke="#000"
        strokeWidth="2"
      />
    </svg>
  </button>
</div>
`;
